packages/core/src/cart/cart-request-sender.ts

```diff
@@ -19,4 +47,139 @@ export default class CartRequestSender {
+
+    loadCard(cartId: string, options: LoadCartRequestOptions): Promise<Response<Cart>> {
+        const url = `/graphql`;
+
+        const graphQLQuery = `
+        query {
+            site {
+                cart(entityId: "${cartId}") {
+                  currencyCode
+                  entityId
+                  id
+                  isTaxIncluded
+                  discounts {
+                    entityId
+                    discountedAmount {
+                      currencyCode
+                      value
+                    }
+                  }
+                  discountedAmount {
+                    currencyCode
+                    value
+                  }
+                  baseAmount {
+                    currencyCode
+                    value
+                  }
+                  amount {
+                    currencyCode
+                    value
+                  }
+                  lineItems {
+                    physicalItems {
+                      brand
+                      couponAmount {
+                        value
+                      }
+                      discountedAmount {
+                        value
+                      }
+                      discounts {
+                        discountedAmount {
+                          value
+                        }
+                        entityId
+                      }
+                      extendedListPrice {
+                        value
+                      }
+                      extendedSalePrice {
+                        value
+                      }
+                      giftWrapping {
+                        amount {
+                          value
+                        }
+                        message
+                        name
+                      }
+                      isShippingRequired
+                      isTaxable
+                      listPrice {
+                        value
+                      }
+                      name
+                      originalPrice {
+                        value
+                      }
+                      entityId
+                      quantity
+                      salePrice {
+                        value
+                      }
+                      sku
+                      url
+                    }
+                  }
+                }
+              }
+        }`;
+
+        const requestOptions: LoadCartRequestOptions = {
+            ...options,
+            headers: {
+                ...options.headers,
+                'Content-Type': 'application/json',
+            },
+            body: {
+                query: graphQLQuery,
+            },
+        };
+
+        return this._requestSender
+            .post<LoadCartResponse>(url, {
+                ...requestOptions,
+            })
+            .then(this.transformToCartResponse);
+    }
+
+    private transformToCartResponse(response: Response<LoadCartResponse>): Response<Cart> {
```

Comment:
@davidchin @animesh1987 @bc-peng guys I would like to ask you to take a loot at this transformer much closer there are concerns I have encountered.

This transformer is going to be very large because here i am trying to transform response in an appropriate Cart interface to align with the current realisation of stored cart information.

There are a few problems from the current approach:

1. it is hard to support/maintain this code in the future
2. we do not have all required Cart fields information for properly transforming
3. double dependency with Cart interface

I would like to suggest creating a new Headless Cart reducer for avoiding bindings to Cart interface and all things above. What do you think?

Reply:
Thanks for bringing this up, my suggestion is that we should not have this in the reducer since that's not the purpose of a reducer. I see this is to transform the response from the loadCart API to a cart here in request sender. 

Maybe we can have something similar to [this](https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/payment/payment-request-transformer.ts) and all transform logic can be placed there. 

Maybe you can also think of moving the transform method in the action creator and not in the request sender.

Reply:
@animesh1987 Why I suggest creating a new reducer is because we can more easily manage this information. This reducer will only take the necessary information for Headless Wallet Buttons strategies

Reply:
Hey @bc-nick what is the GQL endpoint we are calling here? Maybe we can call checkout GQL endpoint for the information you need once you have a cart.

Reply:
@animesh1987 I come back because this question is still relevant. Unfortunately checkout GQL endpoint does not have relevant information for most of required fields. I created the draft [PR](https://github.com/bigcommerce/checkout-sdk-js/pull/2802) where you be able to see all unavailable fields by looking at the TODO messages.